### ISSUE TITLE
no sound in proton 5.13.6 after update

### ISSUE THREAD

**Issue by adcdam** [ValveSoftware/steam-runtime#384]
im using only alsa and not pulseaudio, a few days ago proton 5.13.6 and proton experimental were working fine using just alsa.
after an update i have no sound in proton 5.13.6 and no sound in proton experimental.
please dont tell me ill have to install pulseaudio.
please fix this to be able to work with alsa.

**Comment by RyuzakiKK** [comment]
Can you please provide the Steam's System Information and the `slr-app-*-*.log`? Here you can find the instructions on how to gather them https://github.com/ValveSoftware/steam-runtime/blob/master/doc/reporting-steamlinuxruntime-bugs.md#essential-information

Also, how is your ALSA configured? You edited `/etc/asound.conf` and/or `~/.asoundrc`?

**Comment by adcdam** [comment]
im using the same ~/.asoundrc as before, proton 5.13.6 and proton experimental were working fine with alsa before the last update
how i alsa configured? is not a problem of my system if that is your question.
https://dpaste.com/A36PEFJBQ

**Comment by RyuzakiKK** [comment]
I asked how you configured ALSA because Proton uses the Pressure-Vessel Soldier container and it needs to import the files from the host system that are known to be necessary.

You didn't provide the "Help->System Information" output though and I guess it could give us useful hints to identify the problem.

**Comment by smcv** [comment]
> please dont tell me ill have to install pulseaudio

PulseAudio (or Pipewire pretending to be PulseAudio) is the configuration we recommend: it's what is used for the majority of Steam installations, and the configuration that is going to work best for most people.

If you want to use something else, like direct access to ALSA device nodes, then this is not something we are deliberately breaking, in principle it should work, and we do want it to work - but if that is going to happen, you need to help us. We do not know how your computer is set up, and there is an unlimited number of configurations you *might* be trying to use. If you want your configuration to be one that works, please be as specific about it as possible.

Please show us the Help -> System Information output, and your ALSA configuration.

**Comment by adcdam** [comment]
here you are system information https://github.com/adcdam/s6/blob/master/systeminformation
my home .asoundrc https://github.com/adcdam/s6/blob/master/.asoundrc
my /usr/share/alsa/alsa.conf https://github.com/adcdam/s6/blob/master/alsa.conf
 
the output of aplay -l :                  
**** List of PLAYBACK Hardware Devices ****
card 0: DX [Xonar DX], device 0: Multichannel [Multichannel]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: DX [Xonar DX], device 1: Digital [Digital]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**Comment by smcv** [comment]
If you try the beta branch of *Steam Linux Runtime - soldier*, does that help? We made some changes there that might affect this.

https://github.com/ValveSoftware/steam-runtime/blob/master/doc/reporting-steamlinuxruntime-bugs.md#using-a-beta-or-an-older-version

**Comment by adcdam** [comment]
do i have to download a file from here to do that? https://repo.steampowered.com/steamrt-images-soldier/snapshots/latest-container-runtime-public-beta/  in that case what file should i download?

**Comment by smcv** [comment]
No, please follow the instructions in https://github.com/ValveSoftware/steam-runtime/blob/master/doc/reporting-steamlinuxruntime-bugs.md#using-a-beta-or-an-older-version (it's the same as switching to the beta branch of a game, if you've done that).

**Comment by tall-lantern** [comment]
Hello, I would like to carry the torch forward on this, as it were. 

I did not have success getting sound using the beta branch of `Steam Linux Runtime -soldier`, however I did get sound using the `previous_release` branch. 

I also tried copying my ALSA configuration file (asound.conf) to the runtime `etc` directory, from issue 314, however this was not succesfull. 

Would you like me to submit Sys info and log files?

**Comment by smcv** [comment]
@tall-lantern Please open a separate issue with full details. We don't have enough information to know whether you are experiencing the same issue as @adcdam, or whether you are getting the same symptom for a different reason.

If we resolve one of these issues and it turns out to fix the other one as well, then it's easy to close them both, but if they're different, mixing up two potentially different root causes on one issue number makes it harder to keep track of what is going on, which delays bugs getting fixed.

We will need the logs described in https://github.com/ValveSoftware/steam-runtime/blob/master/doc/reporting-steamlinuxruntime-bugs.md#essential-information and the contents of `steamapps/common/SteamLinuxRuntime_soldier/VERSIONS.txt`, without any special workarounds applied, for at least the branch that is not working.

Since you say there is a previous release where this *does* work for you, it would also be very useful to get the same information from that previous release so we can compare it.

**Comment by tall-lantern** [comment]
Hi @smcv,
I was just going though the process of getting the logs as described in the link, but am now unable to reproduce the issue (no audio by switching to the default branch of `Steam Linux Runtime -soldier`). Both branches now provide audio. Should the issue reoccur, I will immediatly capture the logs and submit a new issue.

**Comment by smcv** [comment]
@tall-lantern wrote:
> I did not have success getting sound using the beta branch of Steam Linux Runtime -soldier, however I did get sound using the previous_release branch

I think I might now see what happened - some code added in soldier v0.20210317.0, which we had intended to only run for PulseAudio users (to fix part of #371), was accidentally also run for non-PulseAudio users (where it's useless, and could break previously-working audio).

If I'm reading the changelog correctly, the possible cause that I've found is most likely to show up as a regression between soldier v0.20210309.0 (worked) and v0.20210317.0 (didn't work).

> am now unable to reproduce the issue (no audio by switching to the default branch of Steam Linux Runtime -soldier). Both branches now provide audio

At the moment there are three interesting branches: `client_beta`, default and `previous_release`. The exact version that is in each branch changes regularly, and might change between the time you write a comment and the time I read it, which is why we always like to see version numbers.

Right now, it looks like soldier v0.20210317.0 is in the beta, v0.20210309.0 is the default version, and something older is in `previous_version` - but that situation might change soon.

Workaround: while we sort this out, if necessary either hold back the version to `previous_release`, or create a `~/.asoundrc` that targets the specific devices you want to be using.

**Comment by smcv** [comment]
The regression reported in https://github.com/ValveSoftware/steam-runtime/issues/384#issuecomment-818873881 is now tracked separately, as #395.

Please use #395 to represent that regression, and only use #384 to represent the original issue reported by @adcdam.
